### ISSUE TITLE
WebODM fails a task if a connection error happens

### ISSUE THREAD

**Issue by pierotofy** [OpenDroneMap/WebODM#727]
In some instances, a task might be flagged as "failed" although NodeODM is still processing the task (and returned an error due to a "hiccup").

This should be fixed by making WebODM more fault tolerant (retry).

A user reported the following for the task failure description:

`Unauthorized. Do you need to set a token?`

Even though NodeODM was still processing.

**Comment by pierotofy** [comment]
Also happens if a process "hangs" for longer than the allowed timeout (happens if the OS starts swapping RAM), freezing the computer state for longer than ~30 seconds.

**Comment by pierotofy** [comment]
Things have stabilized a lot since this was first opened in 2019 and I can't reproduce it. I will assume this is fixed, but we can re-open if the issue happens again.

**Comment by smathermather** [comment]
Agreed. I don't think I have seen it in the last 18 months, even on very long running projecfs.
